modin/config/envvars.py

```diff
@@ -654,6 +654,14 @@ def get(cls) -> int:
         -------
         int
         """
+        from modin.error_message import ErrorMessage
```

Comment:
One potential solution could be to change the class type to tuple, this would localize the partition data in one place and save us from an extra class.

Reply:
By doing this we force the user to specify both sizes at a time - per row and col, while the user may want to specify/adjust a single axis onlt. I would leave as is.

Reply:
By changing this value, the user probably already knows the default values. How else will he start from something to select new values? There should be no problem specifying two values.

Reply:
The user is not forced to know the default values - this is mainly for us to get ultimate performance in general. The user may know the number of rows/cols in advance and may specify fine-grained or coarse-grained partitioning. To me, specifying an additional number looks like extra burden if I don't care about it.

Reply:
Passing two values in a function that doesn't use one of them at all is not a good approach I think.

You suggestion may work for users but it still entails extra burden of calling .get. Also, if the user wants to use the env var, he/she has to pass two values anywat.

Reply:
> Passing two values in a function that doesn't use one of them at all is not a good approach I think.

If we continue the idea of how the shape of a dataframe is stored, then it seems more consistent that its minimum values are stored in the same form.

> Also, if the user wants to use the env var, he/she has to pass two values anyway.

An example of what it would look like:
```python
with context(MinRowPartitionSize=102, MinColumnPartitionSize=16, NPartitions=5):
    ...
# vs
with context(MinPartitionSizes=(102, 16), NPartitions=5):
    ...
```

Reply:
> If we continue the idea of how the shape of a dataframe is stored, then it seems more consistent that its minimum values are stored in the same form.

This may sound reasonable but involves extra burden I don't like.

> An example of what it would look like:

I am talking about the env var, which the user may want to specify outside the script.

Let's ask for other opinions. cc @dchigarev

Reply:
> I am talking about the env var, which the user may want to specify outside the script.

I think the case where we want to specify a row/col partition size outside of the script is the main problem of a single-variable approach, since in this case I somehow need to know what's the value on the opposite axis is:
```
MODIN_MIN_PARTITION_SIZE=(???, 32) python script.py
```

So I would prefer to see them as separate environment variables. However for the sake of convenience, you can actually keep 3 variables: Col/RowPartitionSize and MinPartitionSize taking tuples.

Reply:
> So I would prefer to see them as separate environment variables. However for the sake of convenience, you can actually keep 3 variables: Col/RowPartitionSize and MinPartitionSize taking tuples (assuming that values in all 3 variables would be automatically synced)

too much to me :)

Reply:
> MODIN_MIN_PARTITION_SIZE=(???, 32) python script.py

When running the script in this way, it is enough to look at the default values that we store in the config module. Since the use of these variables is advanced, I take knowledge of the default values for granted, especially given how rarely they can change.

Reply:
We should keep the things exposed to users as much as simple. I think very few people are going to dig into the internals.

Reply:
Without knowing this information, it is not clear how to use these configurations, so I still believe that this is no more complicated than the current approach.

Even with a single environment variable, it's hard to imagine the user specifying a value without looking at the one we use by default. It turns out that user need to look in both situations and in this sense they are equivalent in complexity.

Also default values can be found in https://modin.readthedocs.io/en/latest/flow/modin/config.html#modin-configs-list

Reply:
The main reason by which I think we should keep both MinRow/ColumnPartitionSize is that the user doesn't have to know how to distribute data along a single axis and leave this to Modin itself. Consider a case with a narrow or wide dataframe. If the user has a narrow df, say 10 columns, he/she can set MinColumnPartitionSize to 1 and leave distribution along rows to Modin, which automatically finds a better MinRowPartitionSize if the default one doesn't suit. I stick to keep the current approach.

Reply:
> I stick to keep the current approach.

@YarShev I still think another solution would be better, but since the majority here is for your solution, please update the PR and we'll merge it.

Reply:
done